src/promise-helpers.ts

```diff
@@ -0,0 +1,29 @@
+/*
+  Copyright 2017 Google Inc. All Rights Reserved.
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+export function canvasToBlob(canvas: HTMLCanvasElement, type: string): Promise<Blob> {
+  return new Promise((resolve) => {
+    canvas.toBlob((blob: Blob) => resolve(blob), type);
```

Comment:
You may want to wrap it with `try catch` and reject when an occur occurs.

Reply:
An exception would automatically cause the promise to reject, with the thrown `Error` as the value. You only need to call `reject` explicitly for things that a try/catch block *wouldn't* pick up.

Reply:
I didn't know that. Thank you @wibblymat!